server/src/main/java/org/candlepin/model/Content.java

```diff
@@ -148,6 +173,131 @@ public Content(String id) {
+        this.setContentUrl(source.getContentUrl());
+        this.setRequiredTags(source.getRequiredTags());
+        this.setReleaseVer(source.getReleaseVer());
+        this.setGpgUrl(source.getGpgUrl());
+        this.setMetadataExpire(source.getMetadataExpire());
+        this.setArches(source.getArches());
+
+        this.setModifiedProductIds(source.getModifiedProductIds() != null ?
+            new HashSet<String>(source.getModifiedProductIds()) : null);
+    }
+
+    /**
+     * Copies several properties from the given content on to this content instance. Properties that
+     * are not copied over include any identifiying fields (UUID, ID), the creation date and locking
+     * states. Values on the source content which are null will be ignored.
+     *
+     * @param source
+     *  The source content instance from which to pull content information
+     *
+     * @return
+     *  this content instance
+     */
+    public Content merge(Content source) {
+        this.setUpdated(source.getUpdated() != null ? (Date) source.getUpdated().clone() : null);
+
+        if (source.getType() != null) {
+            this.setType(source.getType());
+        }
+
+        if (source.getLabel() != null) {
+            this.setLabel(source.getLabel());
+        }
+
+        if (source.getName() != null) {
+            this.setName(source.getName());
+        }
+
+        if (source.getVendor() != null) {
+            this.setVendor(source.getVendor());
+        }
+
+        if (source.getContentUrl() != null) {
+            this.setContentUrl(source.getContentUrl());
+        }
+
+        if (source.getRequiredTags() != null) {
+            this.setRequiredTags(source.getRequiredTags());
+        }
+
+        if (source.getReleaseVer() != null) {
+            this.setReleaseVer(source.getReleaseVer());
+        }
+
+        if (source.getGpgUrl() != null) {
+            this.setGpgUrl(source.getGpgUrl());
+        }
+
+        if (source.getMetadataExpire() != null) {
+            this.setMetadataExpire(source.getMetadataExpire());
+        }
+
+        if (source.getArches() != null) {
+            this.setArches(source.getArches());
+        }
+
+        if (source.getModifiedProductIds() != null) {
+            // We need to make a copy of the modified products collection to ensure we don't link
+            // the two instances
+            this.setModifiedProductIds(new HashSet<String>(source.getModifiedProductIds()));
+        }
+
+        return this;
+    }
+
+    @Override
+    public Content clone() {
+        Content copy;
+
+        try {
+            copy = (Content) super.clone();
+        }
+        catch (CloneNotSupportedException e) {
+            // This should never happen.
+            throw new RuntimeException("Clone not supported", e);
+        }
+
+        copy.owners = null;
+        copy.modifiedProductIds = null;
+
+        if (this.getOwners() != null) {
+            copy.setOwners(this.getOwners());
```

Comment:
We've already discussed this in person, but for any other readers, setOwners copies the elements from the given collection into its own collection, so both instances will point to the same owner instances, but won't share a collection instance.